src/query/plan/operator.cpp

```diff
@@ -4024,4 +4026,76 @@ UniqueCursorPtr LoadCsv::MakeCursor(utils::MemoryResource *mem) const {
   return MakeUniqueCursorPtr<LoadCsvCursor>(mem, this, mem);
 };
 
+class ForeachCursor : public Cursor {
+ public:
+  explicit ForeachCursor(Symbol output_symbol, UniqueCursorPtr input, Expression *expr, utils::MemoryResource *mem,
+                         bool is_nested)
+      : output_symbol_(output_symbol),
+        input_cursor_(std::move(input)),
+        expression(expr),
+        cache_(mem),
+        is_nested_(is_nested) {}
+
+  bool Pull(Frame &frame, ExecutionContext &context) override {
+    SCOPED_PROFILE_OP(op_name_);
+
+    if (cache_.empty()) {
+      if (!input_cursor_->Pull(frame, context)) {
+        return false;
+      }
+
+      ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, context.db_accessor,
+                                    storage::View::NEW);
+      TypedValue expr_result = expression->Accept(evaluator);
+      if (!expr_result.IsList()) {
+        throw QueryRuntimeException("FOREACH expression must resolve to a list, but got '{}'.", expr_result.type());
+      }
```

Comment:
Don't we want to handle the null values in a nice way? 

```
CREATE (n);
MATCH (n) FOREACH ( li IN n.prop | CREATE (:F4 {prop: li}) CREATE (:F5 {prop2: li}) SET n.li = li);
```
Does notthing in neo4j, but throws and exception in our case. I think this is a nice little addition that really makes users' life easier.